x/datadeal/keeper/grpc_query_deal.go

```diff
@@ -24,17 +26,79 @@ func (k Keeper) Deal(goCtx context.Context, req *types.QueryDealRequest) (*types
 	}, nil
 }
 
-func (k Keeper) Deals(ctx context.Context, request *types.QueryDealsRequest) (*types.QueryDealsResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (k Keeper) Deals(goCtx context.Context, req *types.QueryDealsRequest) (*types.QueryDealsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	store := ctx.KVStore(k.storeKey)
+	dealsStore := prefix.NewStore(store, types.DealKey)
+
+	var deals []*types.Deal
+	pageRes, err := query.Paginate(dealsStore, req.Pagination, func(_ []byte, value []byte) error {
+		var deal types.Deal
+		err := k.cdc.UnmarshalLengthPrefixed(value, &deal)
+		if err != nil {
+			return err
+		}
+		deals = append(deals, &deal)
+		return nil
+	})
+
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &types.QueryDealsResponse{
+		Deals:      deals,
+		Pagination: pageRes,
+	}, nil
 }
 
-func (k Keeper) Certificates(ctx context.Context, certificates *types.QueryCertificates) (*types.QueryCertificatesResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (k Keeper) Certificates(goCtx context.Context, req *types.QueryCertificates) (*types.QueryCertificatesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	store := ctx.KVStore(k.storeKey)
+	certsStore := prefix.NewStore(store, types.CertificateKey)
```

Comment:
The deal ID probably should be included in prefix.

Reply:
Thanks. fixed it(https://github.com/medibloc/panacea-core/pull/539/commits/0818bb2f49fb664dc8d8434ed545e68157e90b6a)